src/pages/index.tsx

```diff
@@ -4,35 +4,33 @@ import { PokemonInfo } from "@/models/PokemonInfo";
 import "./index.css";
 import { useEffect, useState } from "react";
 import { NamedAPIResource, PokemonClient } from "pokenode-ts";
+import { IndexNumberSearch } from "@/components/IndexNumberSearch";
 
 export default function Home() {
   const [allPokemons, setAllPokemons] = useState<PokemonInfo[]>([]);
 
-  const getAllPokemons = async () => {
-    const api = new PokemonClient();
-    const pokemons = await api.listPokemons(0, 20);
-    pokemons.results.forEach((p) => {
-      console.log(p.name);
-    });
-    await createPokemonObject(pokemons.results);
+  const onClickSearch = async (val: number) => {
+    setAllPokemons([]);
+    await fetchTwentyPokemons(val);
   };
 
-  const createPokemonObject = async (results: NamedAPIResource[]) => {
+  const fetchTwentyPokemons = async (offset: number = 1) => {
     const api = new PokemonClient();
-    for (const res of results) {
-      const pokemon = await api.getPokemonByName(res.name);
+    for (let i = 0; i < 20; i++) {
+      const pokemon = await api.getPokemonById(i + offset);
       const newList: PokemonInfo = {
         id: pokemon.id,
         name: pokemon.name,
         image: pokemon.sprites.other!["official-artwork"].front_default!,
         type: pokemon.types[0].type.name,
       };
+
       setAllPokemons((currentList) => [...currentList, newList]);
```

Comment:
[imo] こんな感じにすれば並列にリクエストできるから早くなるかも

```suggestion
const promises = Array.from({ length: 20 }, (_, i) => 
  api.getPokemonById(i + offset).then(pokemon => ({
    id: pokemon.id,
    name: pokemon.name,
    image: pokemon.sprites.other!["official-artwork"].front_default!,
    type: pokemon.types[0].type.name,
  }))
);

 const results = await Promise.all(promises);
```